changelogs/2.10.0.md

```diff
@@ -0,0 +1,933 @@
+
+## bugfix/raft
+
+* Fixed `box.ctl.promote()` entering an infinite election loop when a node
+  doesn't have enough peers to win the elections (gh-6654).
+* Servers with elections enabled will resign the leadership and become
+  read-only when the number of connected replicas becomes less than a quorum.
+  This should prevent split-brain in some situations (gh-6661).
+
+### LuaJIT
+
+* Introduced support for `LJ_DUALNUM` mode in luajit-gdb.py (gh-6224).
+* Introduced preliminary support of GNU/Linux ARM64 and MacOS M1. In the scope
+  of this activity, the following issues have been resolved:
+
+  - Introduced support for a full 64-bit range of lightuserdata values (gh-2712).
+
+  - Fixed memory remapping issue when the page leaves 47-bit segments.
+
+  - Fixed M1 architecture detection (gh-6065).
+
+  - Fixed variadic arguments handling in FFI on M1 (gh-6066).
+
+  - Fixed `table.move` misbehavior when table reallocation occurs (gh-6084).
+
+  - Fixed Lua stack inconsistency when xpcall is called with an invalid second
+    argument on ARM64 (gh-6093).
+
+  - Fixed `BC_USETS` bytecode semantics for closed upvalues and gray strings.
+
+  - Fixed side exit jump target patching considering the range values of the
+    particular instruction (gh-6098).
+
+  - Fixed current Lua coroutine restoring on an exceptional path on ARM64 (gh-6189).
+* Now, memory profiler records allocations from traces grouping them by the
+  trace number (gh-5814). The memory profiler parser can display the new type
+  of allocation sources in the following format:
+
+  ```
+  | TRACE [<trace-no>] <trace-addr> started at @<sym-chunk>:<sym-line>
+  ```
+* Now memory profiler reports allocations made by the JIT engine while
+  compiling the trace as INTERNAL (gh-5679).
+* Now the memory profiler emits events of the new type when a function or
+  a trace is created. As a result the memory profiler parser can enrich its
+  symbol table with the new functions and traces (gh-5815).
+
+  Furthermore, there are symbol generations introduced within the internal
+  parser structure to handle possible collisions of function addresses and
+  trace numbers.
+* Now memory profiler dumps symbol table for C functions. As a result memory
+  profiler parser can enrich its symbol table with C symbols (gh-5813).
+  Furthermore, now memprof dumps special events for symbol table when it
+  encounters a new C symbol, that hasn't been dumped yet.
+* Introduced the LuaJIT platform profiler (gh-781) and the profile parser.
+This profiler is able to capture both host and VM stacks, so it can show the
+whole picture. Both C and Lua API's are available for the profiler.<br/>
+Profiler comes with the default parser, which produces output in a
+flamegraph.pl-suitable format.<br/>
+The following profiling modes are available:
+  - Default: only virtual machine state counters
+  - Leaf: shows the last frame on the stack
+  - Callchain: performs a complete stack dump
```

Comment:
```suggestion
* Introduced the LuaJIT platform profiler (gh-781) and the profile parser.
  This profiler is able to capture both host and VM stacks, so it can show the
  whole picture. Both C and Lua API's are available for the profiler.<br/>
  Profiler comes with the default parser, which produces output in a
  flamegraph.pl-suitable format.<br/>
  The following profiling modes are available:
  - Default: only virtual machine state counters
  - Leaf: shows the last frame on the stack
  - Callchain: performs a complete stack dump
```